packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -225,15 +230,25 @@ export class VSXExtensionsModel {
                 if (!allVersions) {
                     continue;
                 }
-                this.setExtension(id).update(Object.assign(data, {
-                    publisher: data.namespace,
-                    downloadUrl: data.files.download,
-                    iconUrl: data.files.icon,
-                    readmeUrl: data.files.readme,
-                    licenseUrl: data.files.license,
-                    version: allVersions.version
-                }));
-                searchResult.add(id);
+                const res = await client.query({ extensionId: id, extensionVersion: allVersions.version, includeAllVersions: true });
```

Comment:
We're doing an extra query for each extension. Can't we get this info from the initial query? If not, let's open an issue a https://github.com/eclipse/openvsx to either be able to specify "verified" as a criterion or to get the flag in the search result.

Reply:
During the search, we are using the Search API (https://open-vsx.org/swagger-ui/index.html#/registry-api/search) to get the list of extensions but they do not include the `verified` flag, which is why I had to do this extra query.

I have opened an issue: https://github.com/eclipse/openvsx/issues/824

Thanks for the feedback.

Reply:
Thanks for opening the issue, it would be by far the best solution to just do this in the original search.